src/routes/Onboarding/utils.tsx

```diff
@@ -70,69 +86,94 @@ export function useDemoCode() {
+        `Failed to navigate to invalid onboarding status ${newStatus}`
+      )
+    }
     settingsActor.send({
       type: 'set.app.onboardingStatus',
       data: { level: 'user', value: newStatus },
     })
-    navigate(filePath + PATHS.ONBOARDING.INDEX.slice(0, -1) + newStatus)
-  }, [filePath, newStatus, settingsActor.send, navigate])
+    navigate(joinRouterPaths(filePath, PATHS.ONBOARDING.INDEX, newStatus))
+  }, [filePath, newStatus, navigate])
 }
 
 export function useDismiss() {
   const filePath = useAbsoluteFilePath()
   const send = settingsActor.send
   const navigate = useNavigate()
 
-  const settingsCallback = useCallback(() => {
-    send({
-      type: 'set.app.onboardingStatus',
-      data: { level: 'user', value: 'dismissed' },
-    })
-    waitFor(settingsActor, (state) => state.matches('idle'))
-      .then(() => navigate(filePath))
-      .catch(reportRejection)
-  }, [send])
+  const settingsCallback = useCallback(
+    (
+      dismissalType:
+        | Extract<OnboardingStatus, 'completed' | 'dismissed'>
+        | undefined = 'dismissed'
+    ) => {
+      send({
+        type: 'set.app.onboardingStatus',
+        data: { level: 'user', value: dismissalType },
+      })
+      waitFor(settingsActor, (state) => state.matches('idle'))
+        .then(() => {
+          navigate(filePath)
+          toast.success(
+            'Click the question mark in the lower-right corner if you ever want to redo the tutorial!',
+            {
+              duration: 5_000,
+            }
+          )
+        })
+        .catch(reportRejection)
+    },
+    [send, filePath, navigate]
+  )
 
   return settingsCallback
 }
+
 export function OnboardingButtons({
   currentSlug,
   className,
   dismissClassName,
   onNextOverride,
   ...props
 }: {
-  currentSlug?: (typeof onboardingPaths)[keyof typeof onboardingPaths]
+  currentSlug?: (typeof ONBOARDING_SUBPATHS)[keyof typeof ONBOARDING_SUBPATHS]
   className?: string
   dismissClassName?: string
   onNextOverride?: () => void
 } & React.HTMLAttributes<HTMLDivElement>) {
+  const onboardingPathsArray = Object.values(ONBOARDING_SUBPATHS)
   const dismiss = useDismiss()
   const stepNumber = useStepNumber(currentSlug)
   const previousStep =
-    !stepNumber || stepNumber === 0 ? null : onboardingRoutes[stepNumber - 2]
-  const goToPrevious = useNextClick(
-    onboardingPaths.INDEX + (previousStep?.path ?? '')
-  )
+    !stepNumber || stepNumber <= 1 ? null : onboardingPathsArray[stepNumber]
   const nextStep =
-    !stepNumber || stepNumber === onboardingRoutes.length
+    !stepNumber || stepNumber === onboardingPathsArray.length
       ? null
-      : onboardingRoutes[stepNumber]
-  const goToNext = useNextClick(onboardingPaths.INDEX + (nextStep?.path ?? ''))
+      : onboardingPathsArray[stepNumber]
+
+  const previousOnboardingStatus: OnboardingStatus =
+    previousStep ?? ONBOARDING_SUBPATHS.INDEX
+  const nextOnboardingStatus: OnboardingStatus = nextStep ?? 'completed'
```

Comment:
In #6564 this line was
```ts
  const nextOnboardingStatus: OnboardingStatus = nextStep + ONBOARDING_SUBPATHS.INDEX
```

For some inexplicable reason. The `OnboardingStatus` type def added provides red squiggles in development, in addition to the runtime error handling mentioned elsewhere.